File: src/main/java/com/anatartari/tatamecontrolapi/infra/database/entity/StudentEntity.java
package com.anatartari.tatamecontrolapi.infra.database.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@Entity
@Table(name = "student")
public class StudentEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false)
    private String phone;

    private String gender;

    @Column(nullable = false)
    private LocalDate birthday;

    @Column(columnDefinition = "boolean default false")
    private Boolean allowSocialMedia;

    private String instagram;

    private Boolean practicedMartialArts;

    private String graduatedInStyle;

    @OneToOne(fetch =  FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "address_id", referencedColumnName = "id")
    private AddressEntity address;

    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "medical_info_id", referencedColumnName = "id")
    private MedicalInfoEntity medicalInfo;
}

File: src/student/domain/entities/student.entity.ts
import { Entity, Column, ManyToOne, JoinColumn } from 'typeorm';
import { BaseEntity } from '../../../global/entities/base.entity';
import { Address } from '../../../address/domain/entities/address.entity';
import { MedicalInfo } from '../../../medical-info/domain/entities/medical-info.entity';

@Entity()
export class Student extends BaseEntity {

    @Column({ length: 255 })
    name: string;

    @Column({ length: 255, unique: true })
    email: string;

    @Column({ length: 20 })
    phone: string;

    @Column({ length: 50, nullable: true })
    gender?: string;

    @Column({ type: 'date' })
    birthday: Date;

    @Column({ name: 'allowSocialMedia', default: false })
    allowSocialMedia: boolean;

    @Column({ length: 255, nullable: true })
    instagram?: string;

    @Column({ name: 'practicedMartialArts', nullable: true })
    practicedMartialArts?: boolean;

    @Column({ name: 'graduatedInStyle', length: 255, nullable: true })
    graduatedInStyle?: string;

    @ManyToOne(() => Address, { onDelete: 'SET NULL' })
    @JoinColumn({ name: 'addressId' })
    address?: Address;

    @ManyToOne(() => MedicalInfo, { onDelete: 'SET NULL' })
    @JoinColumn({ name: 'medicalInfoId' })
    medicalInfo?: MedicalInfo;
}

